draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -8403,7 +8403,7 @@ Content-Range: exampleunit 11.2-14.3/25
 </t>
 <t>
    A 204 response is terminated by the first empty line after the header
-   fields because it cannot contain a message body.
+   fields because it cannot contain a message body or trailers.
 </t>
```

Comment:
that seems to be a 11ism

Reply:
I was hoping to avoid this; there's a much deeper question here as to whether a response without a body (1xx, 204 and 304) can have trailers, regardless of HTTP version. Should we address it?

Reply:
I guess to. It is possible in HTTP/2, right?

Reply:
In the sense that a GET can have a body, possibly. Not sure what implementations do, though.

Reply:
The actual point I was trying to make is that the whole statement ("...terminated by first empty line...") is specific to HTTP/1.1.

Reply:
Ah, good point, will push a correction.

Reply:
For me there's a big issue there that needs to be addressed, I agree. 7540 is silent on this point. It doesn't deny it and technically it's easy to do it. But does it have any sense at all if we consider that historically the trailers were anything post-body ?  But I can understand why some implementations would like to systematically put a status in trailers, not only to ease processing, but also because it's not always easy to guess the result before sending headers. I think grpc mandates something like this for the grpc-status trailers, though I'm not sure what it suggests for bodyless responses.

Reply:
I think the root of the question revolves around "what should an agent do when it receives a HEADERS frame indicating there's no body (204/304/response to HEAD), DATA frames carrying a body, and a HEADERS frame containing trailers. Should it just silently ignore the body but still parse the trailers ? Should it stop after headers ? Should it error ? But in H1 we didn't error, we only had to possibly close the connection after headers in such a case (if it was ever detected).